mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list.go

```diff
@@ -0,0 +1,414 @@
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+
+	r.client = client
+}
+
+func (r *ProjectIPAccessListRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"cidr_block": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					cstmvalidator.ValidCIDR(),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("aws_security_group"),
+						path.MatchRelative().AtParent().AtName("ip_address"),
+					}...),
+				},
+			},
+			"ip_address": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					cstmvalidator.ValidIP(),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("aws_security_group"),
+						path.MatchRelative().AtParent().AtName("cidr_block"),
+					}...),
+				},
+			},
+			"aws_security_group": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("ip_address"),
+						path.MatchRelative().AtParent().AtName("cidr_block"),
+					}...),
+				},
+			},
+			"comment": schema.StringAttribute{
+				Computed: true,
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Delete: true,
+				Read:   true,
+			}),
+		},
+	}
+}
+
+func (r *ProjectIPAccessListRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var projectIPAccessListModel *tfProjectIPAccessListModel
+
+	diags := req.Plan.Get(ctx, &projectIPAccessListModel)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if projectIPAccessListModel.CIDRBlock.IsNull() && projectIPAccessListModel.IPAddress.IsNull() && projectIPAccessListModel.AWSSecurityGroup.IsNull() {
+		resp.Diagnostics.AddError("validation error", "cidr_block, ip_address or aws_security_group needs to contain a value")
+		return
+	}
+
+	conn := r.client.Atlas
+	projectID := projectIPAccessListModel.ProjectID.ValueString()
+	stateConf := &retry.StateChangeConf{
```

Comment:
[q] any changes/issues when migrating `StateChangeConf`? Or was this piece migrated as is from previous implementations?

Reply:
yes, this was moved from the previous implementation https://github.com/mongodb/terraform-provider-mongodbatlas/blob/af8d75bc7f81d2073931e776cba3fa9059cf2d5d/mongodbatlas/resource_mongodbatlas_project_ip_access_list.go#L108